elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -279,6 +279,13 @@ public FilterExpression visit(ComparisonNode node, Class entityType) {
                     .map(Path.PathElement::getFieldType)
                     .orElseThrow(() -> new IllegalStateException("Path must not be empty"));
 
+            if (Number.class.isAssignableFrom(relationshipType)) {
+                //Support number types
+                for (int i = 0; i < arguments.size(); i++) {
+                    arguments.set(i, arguments.get(i).replace("*", ""));
```

Comment:
Stripping this here means we lose the context of what operator is being applied.  We need to know if this is startsWith, endsWith, contains, or equals.  The position of the asterisk (beginning, end, or both beginning and end) defines which operator applies.